test/powershell/engine/Remoting/SSHRemotingAPI.Tests.ps1

```diff
@@ -1,41 +1,68 @@
+Import-Module $PSScriptRoot\..\..\Common\Test.Helpers.psm1
+
 Describe "SSH Remoting API Tests" -Tags "Feature" {
 
     Context "SSHConnectionInfo Class Tests" {
 
         It "SSHConnectionInfo constructor should throw null argument exception for null HostName parameter" {
 
+            { [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
+                "UserName",
+                [System.Management.Automation.Internal.AutomationNull]::Value,
+                [System.Management.Automation.Internal.AutomationNull]::Value,
+                0) } | ShouldBeErrorId "PSArgumentNullException"
+        }
+
+        It "SSHConnectionInfo should throw file not found exception for invalid key file path" {
+
             try
             {
-                [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
+                $sshConnectionInfo = [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
                     "UserName",
-                    [System.Management.Automation.Internal.AutomationNull]::Value,
-                    [System.Management.Automation.Internal.AutomationNull]::Value)
+                    "localhost",
+                    "NoValidKeyFilePath",
+                    22)
 
+                $rs = [runspacefactory]::CreateRunspace($sshConnectionInfo)
+                $rs.Open()
+                
                 throw "No Exception!"
             }
             catch
             {
-                $_.FullyQualifiedErrorId | Should Be "PSArgumentNullException"
+                $expectedFileNotFoundException = $_.Exception
+                if ($_.Exception.InnerException -ne $null -and $_.Exception.InnerException.InnerException -ne $null)
+                {
+                    $expectedFileNotFoundException = $_.Exception.InnerException.InnerException
+                }
+                
+                $expectedFileNotFoundException | Should BeOfType "System.IO.FileNotFoundException"
```

Comment:
I believe that there is no need to examine the properties, we can get an undesirable exception only when a method is called (ex., $prop.GetType()).
Here we can make the test more graceful and use `ShouldBeErrorId` too:
```powershell
$exc = {
                $sshConnectionInfo = [System.Management.Automation.Runspaces.SSHConnectionInfo]::new(
                    "UserName",
                    "localhost",
                    "NoValidKeyFilePath",
                    22)

                $rs = [runspacefactory]::CreateRunspace($sshConnectionInfo)
                $rs.Open()
} | ShouldBeErrorId "PSRemotingDataStructureException"

$exc.Exception.InnerException.InnerException | Should BeOfType System.IO.FileNotFoundException
```
Also the tests skip a cleanup code. Please add something like:
```powershell
        AfterEach {
            if ($rs -ne $null) { $rs.Dispose() }
        }
```